src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs

```diff
@@ -97,15 +97,16 @@ internal ProxyDataCollectionManager(IRequestData requestData, string settingsXml
         /// <param name="dataCollectionLauncher">
         ///     Launches datacollector process.
         /// </param>
-        internal ProxyDataCollectionManager(IRequestData requestData, string settingsXml,
+        internal ProxyDataCollectionManager(IRequestData requestData, TestRunCriteria testRunCriteria,
             IDataCollectionRequestSender dataCollectionRequestSender, IProcessHelper processHelper,
             IDataCollectionLauncher dataCollectionLauncher)
         {
             // DataCollector process needs the information of the Extensions folder
             // Add the Extensions folder path to runsettings.
-            this.settingsXml = UpdateExtensionsFolderInRunSettings(settingsXml);
+            this.testRunCriteria = testRunCriteria;
+            var settingsXml = UpdateExtensionsFolderInRunSettings(testRunCriteria.TestRunSettings);
+            this.testRunCriteria = new TestRunCriteria(testRunCriteria.Sources, testRunCriteria.FrequencyOfRunStatsChangeEvent, testRunCriteria.KeepAlive, settingsXml);
```

Comment:
Don't make new testRunCriteria. Let's go with separate settingsXml and testSources.